endpoints/v2/manifest.py

```diff
@@ -405,6 +405,38 @@ def delete_manifest_by_digest(namespace_name, repo_name, manifest_ref):
         return Response(status=202)
 
 
+@v2_bp.route(MANIFEST_TAGNAME_ROUTE, methods=["DELETE"])
+@disallow_for_account_recovery_mode
+@parse_repository_name()
+@process_registry_jwt_auth(scopes=["pull", "push"])
+@log_unauthorized_delete
+@require_repo_write(allow_for_superuser=True, disallow_for_restricted_users=True)
+@anon_protect
+@check_readonly
+@check_pushes_disabled
+def delete_manifest_by_tag(namespace_name, repo_name, manifest_ref):
```

Comment:
This won't necessarily delete the manifest right? Only the pointer. i.e You could delete the tag pointing to a manifest but the underlying manifest would still exist if it's referenced by another tag, has a time machine, ...

Reply:
No, it will only remove the tag pointing to that SHA digest, leaving the SHA digest intact:

```
# curl -H "Authorization: Bearer eyJhbG..." https://quay.skynet/v2/ibazulic/httpd-24/tags/list
{"name":"ibazulic/httpd-24","tags":["latest-new-image","new-tag-added"]}
#
```

Both of these are pointing to the same SHA digest:

```
# skopeo inspect docker://quay.skynet/ibazulic/httpd-24:latest-new-image | jq '.Digest'
"sha256:579faed2d24b32532927720aa9f40021d168cf1a59e49584b7c5c5913cf0accb"
# skopeo inspect docker://quay.skynet/ibazulic/httpd-24:new-tag-added | jq '.Digest'
"sha256:579faed2d24b32532927720aa9f40021d168cf1a59e49584b7c5c5913cf0accb"
```

After deletion, only one tag remains:

~~~
# curl -X DELETE -H "Authorization: Bearer eyJhb..." "https://quay.skynet/v2/ibazulic/httpd-24/manifests/new-tag-added"
# curl -H "Authorization: Bearer eyJhbG..." https://quay.skynet/v2/ibazulic/httpd-24/tags/list
{"name":"ibazulic/httpd-24","tags":["latest-new-image"]}
# podman pull quay.skynet/ibazulic/httpd-24:new-tag-added
Trying to pull quay.skynet/ibazulic/httpd-24:new-tag-added...
WARN[0000] Failed, retrying in 1s ... (1/3). Error: initializing source docker://quay.skynet/ibazulic/httpd-24:new-tag-added: reading manifest new-tag-added in quay.skynet/ibazulic/httpd-24: unknown: Tag new-tag-added was deleted or has expired. To pull, revive via time machine 
...
# podman pull quay.skynet/ibazulic/httpd-24:latest-new-image
Trying to pull quay.skynet/ibazulic/httpd-24:latest-new-image...
Getting image source signatures
Copying blob 9414838f6391 done   | 
Copying blob 051630b2d0a7 done   | 
Copying blob ebbef44f9449 done   | 
Copying config 1723ee8efe done   | 
Writing manifest to image destination
1723ee8efe3e5628747775cf120d783b0eecd6cca299837fae64f9f867b5eb9a
~~~

The tag is in the time machine till it expires. At that point, the tag gets removed, but content still linked remains. Example:

~~~
# curl -H "Authorization: Bearer eyJhb..." https://quay.skynet/v2/ibazulic/httpd-24/tags/list
{"name":"ibazulic/httpd-24","tags":["latest-new-image","new-tag-delete-test-with-0-seconds-time-machine"]}
~~~

These two tags are pointing to the same digest, as before. After deletion with time machine setting to 0 seconds:

~~~
# curl -H "Authorization: Bearer eyJhbG..." https://quay.skynet/v2/ibazulic/httpd-24/tags/list
{"name":"ibazulic/httpd-24","tags":["latest-new-image"]}
# podman pull quay.skynet/ibazulic/httpd-24:latest-new-image
Trying to pull quay.skynet/ibazulic/httpd-24:latest-new-image...
Getting image source signatures
Copying blob 051630b2d0a7 done   | 
Copying blob 9414838f6391 done   | 
Copying blob ebbef44f9449 done   | 
Copying config 1723ee8efe done   | 
Writing manifest to image destination
1723ee8efe3e5628747775cf120d783b0eecd6cca299837fae64f9f867b5eb9a
~~~

So content stays preserved in all cases.